packages/server/src/api/controllers/query/import/sources/openapi3.ts

```diff
@@ -0,0 +1,189 @@
+import { ImportInfo } from "./base"
+import { Query, QueryParameter } from "../../../../../definitions/datasource"
+import { OpenAPIV3 } from "openapi-types"
+import { OpenAPISource } from "./base/openapi"
+import { URL } from "url"
+
+const parameterNotRef = (
+  param: OpenAPIV3.ParameterObject | OpenAPIV3.ReferenceObject
+): param is OpenAPIV3.ParameterObject => {
+  // all refs are deferenced by parser library
+  return true
+}
+
+const requestBodyNotRef = (
+  param: OpenAPIV3.RequestBodyObject | OpenAPIV3.ReferenceObject | undefined
+): param is OpenAPIV3.RequestBodyObject => {
+  // all refs are deferenced by parser library
+  return param !== undefined
+}
+
+const schemaNotRef = (
+  param: OpenAPIV3.SchemaObject | OpenAPIV3.ReferenceObject | undefined
+): param is OpenAPIV3.SchemaObject => {
+  // all refs are deferenced by parser library
+  return param !== undefined
+}
+
+const isOpenAPI3 = (document: any): document is OpenAPIV3.Document => {
+  return document.openapi.includes("3")
```

Comment:
There are some breaking changes to the spec in 3.1. Probably better we be more specific here with `includes("3.0")` instead